### PR TITLE
Implement an rpi-eep crate and an example application [REVPI-2447]

### PR DESCRIPTION
**Pull request by iluminat23**

### CODE REVIEW

**Comment by ferdrich-kunbus** [comment]
Typo in commit heading [bcf1471](https://github.com/RevolutionPi/revpi-hat-eeprom/pull/7/commits/bcf1471562ad3a21aa4b34c628838422d2850639) -> indentation

File: rpi-eep/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+pub struct EEPAtom {
+    atype: EEPAtomType,
+    /// incrementing atom count
+    count: u16,
+    // length in bytes of data+CRC
+    //dlen: u32,
+    /// N bytes, N = dlen-2
+    data: EEPAtomData,
+    // CRC-16 of entire atom (type, count, dlen, data)
+    //crc16: u16,
```
Comment (ferdrich-kunbus) [inline rpi-eep/src/lib.rs:189]: dlen and crc16 are not needed? will they be added in a future update?
If not needed, please add a comment, why.

File: rpi-eep/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+        let dlen = self.data.len() as u32 + 2;
+        buf.extend_from_slice(&dlen.to_le_bytes());
+        self.data.to_bytes(buf);
+
+        let crc_len = self.len() - 2;
+        let crc16 = ATOM_CRC16.checksum(&buf[(buf.len() - crc_len)..]);
+        buf.extend_from_slice(&crc16.to_le_bytes());
+    }
+}
+/// This struct implements the Vendor info Atom
```
Comment (ferdrich-kunbus) [inline rpi-eep/src/lib.rs:247]: missing Newline

File: rpi-eep/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+/// [Atom Structure](https://github.com/raspberrypi/hats/blob/9616b5cd2bdf3e1d2d0330611387d639c1916100/eeprom-format.md#atom-structure):
+/// ```text
+/// Bytes   Field
+/// 2       type        atom type
+/// 2       count       incrementing atom count
+/// 4       dlen        length in bytes of data+CRC
+/// N       data        N bytes, N = dlen-2
+/// 2       crc16       CRC-16 of entire atom (type, count, dlen, data)
+/// ```
+#[derive(Debug)]
```
Comment (ferdrich-kunbus) [inline rpi-eep/src/lib.rs:179]: Mixed usage of inline documentation and documentation bevore the declaration of a struct or eeprom. I don't know when to use which variant, but is there any logic behind?

**Comment by ferdrich-kunbus** [comment]
Which crate is meaned in the commit message of commit [3c9cff9](https://github.com/RevolutionPi/revpi-hat-eeprom/pull/7/commits/3c9cff9fdcef79e03c042e461f401ad73755bc7f) with revpi-hat-eeprom?

**Review by ferdrich-kunbus — CHANGES_REQUESTED** [review]
Looks almost ok with some small hickups, typos and personal understanding questions.
Please fix at least the typos.

File: rpi-eep/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# SPDX-FileCopyrightText: Copyright KUNBUS GmbH
+
+[package]
+name = "rpi-eep"
```
[the remark this answers is not in the document]
Comment (iluminat23) [reply rpi-eep/Cargo.toml:5]: Done

File: revpi-eep/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# SPDX-FileCopyrightText: Copyright KUNBUS GmbH
+
+[package]
+name = "revpi-eep"
```
[the remark this answers is not in the document]
Comment (iluminat23) [reply revpi-eep/Cargo.toml:5]: dito

File: rpi-eep/src/lib.rs (hunk 1 above)
Comment (iluminat23) [reply rpi-eep/src/lib.rs:189]: done

File: rpi-eep/src/lib.rs (hunk 3 above)
Comment (iluminat23) [reply rpi-eep/src/lib.rs:179]: One is just copied verbatim from the the original documentation. The other documents the properties. The Whole documentation needs to be polished. I will leave this for now as is.

**Review by ferdrich-kunbus — APPROVED** [review]
Ready for merge

**Pull request merged**